### PR TITLE
Add timeouts into the SDK

### LINKED ISSUE #79: SDK should handle timeouts
Right now in the client we use Qt machinery to handle timeouts for API calls. This ends up being rather complex. It would be more convenient to handle timeouts in the SDK so that methods that call the API simple raise `ApiTimeoutError`s instead.

### PR DESCRIPTION
**Pull request by heartsucker**
Fixes #79

- Creates a timeout exception to catch all possible timeouts from `requests` / Qubes RPC
- Remove mutable global state for the proxy VM name
- Minor refactor to increase readability of code

### CODE REVIEW

File: sdclientapi/__init__.py [hunk 1]
```diff
@@ -17,33 +18,44 @@
...
-    return output.strip()
+
+    try:
+        stdout, _ = p.communicate(timeout=timeout)  # type: (bytes, bytes)
+    except TimeoutExpired:
+        try:
+            p.terminate()
+        except Exception:
+            pass
+        raise RequestTimeoutError
```
Comment (redshiftzero) [inline sdclientapi/__init__.py]: we should add additional tests to exercise this path, i.e. verifying that the timeout exception is raised when a request takes longer than `DEFAULT_REQUEST_TIMEOUT`

Comment (heartsucker) [reply sdclientapi/__init__.py]: fixed

File: sdclientapi/__init__.py [hunk 2]
```diff
@@ -480,7 +562,9 @@ def download_submission(
         :returns: Tuple of sha256sum and path of the saved submission.
         """
         path_query = "api/v1/sources/{}/submissions/{}/download".format(
-            submission.source_uuid, submission.uuid
+            submission.source_uuid,
+            submission.uuid,
+            timeout=timeout or self.default_download_timeout,
```
Comment (creviera) [inline sdclientapi/__init__.py]: Does this work? Won't `timeout` get ignored since there isn't a `{timeout}` var in the string?

Comment (creviera) [reply sdclientapi/__init__.py]: Also do we want to pass a timeout to `download_reply` the way we do here for `download_submission`?

Comment (heartsucker) [reply sdclientapi/__init__.py]: Oh wow, I added that to the wrong place entirely. Good catch.

Comment (heartsucker) [reply sdclientapi/__init__.py]: Also I figured it's ok for now to not pass this to replies since realistically we aren't going to have huge replies.

**Pull request merged**